### PR TITLE
Add fused_feed_forward pass

### PR DESCRIPTION
**Pull request by GhostScreaming**
### PR types
Others

### PR changes
Others

### Describe
Add fused_feed_forward pass for semi-automatic static graph training.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_fused_feedforward_pass.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            loss = paddle.mean(out)
+            sgd_optimizer = paddle.fluid.optimizer.SGD(learning_rate=0.001)
+            sgd_optimizer.minimize(loss)
+
+        if use_pass:
+            pass_manager = PassManager([new_pass("fused_feedforward")])
+            pass_manager.apply([main_prog], [startup_prog])
+
+            ops = main_prog.global_block().ops
+            assert ops[2].type == 'fused_feedforward'
```
Comment (kuizhiqing) [inline python/paddle/fluid/tests/unittests/test_fused_feedforward_pass.py]: Is `in` already ok?

File: paddle/fluid/framework/ir/fused_feedforward_pass.cc [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+        "dropout2_rate", dropout_op_grad_2->Op()->GetAttr("dropout_prob"));
+    fused_feedforward_op_desc.SetAttr(
+        "dropout1_implementation",
+        dropout_op_grad_1->Op()->GetAttr("dropout_implementation"));
+    fused_feedforward_op_desc.SetAttr(
+        "dropout2_implementation",
+        dropout_op_grad_2->Op()->GetAttr("dropout_implementation"));
+    fused_feedforward_op_desc.SetAttr("add_residual", add_residual);
+    // These attributes set default value
+    fused_feedforward_op_desc.SetAttr("is_test", false);
```
Comment (kuizhiqing) [inline paddle/fluid/framework/ir/fused_feedforward_pass.cc]: Checkout is always False is OK?

File: paddle/fluid/framework/ir/fused_feedforward_pass.cc [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+    fused_feedforward_op_desc.SetAttr("dropout1_fix_seed", false);
+    fused_feedforward_op_desc.SetAttr("dropout2_fix_seed", false);
+    fused_feedforward_op_desc.SetAttr("dropout1_seed", 0);
+    fused_feedforward_op_desc.SetAttr("dropout2_seed", 0);
+    fused_feedforward_op_desc.SetAttr("ring_id", -1);
```
Comment (kuizhiqing) [inline paddle/fluid/framework/ir/fused_feedforward_pass.cc]: TODO

**Review by kuizhiqing — APPROVED** [review]
lgtm

**Review by sneaxiy — APPROVED** [review]
LGTM. Excellent job.

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by kolinwei — APPROVED** [review]

**Pull request merged**